libmamba/src/core/run.cpp

```diff
@@ -470,12 +519,23 @@ namespace mamba
 
             ec = reproc::drain(proc, reproc::sink::null, reproc::sink::null);
 
-            std::tie(status, ec) = proc.stop(opt.stop);
+#ifndef _WIN32
+            status = wait_for_process(pid);
```

Comment:
Between the place where we installed the signal handlers up until after this line, is it possible to get any exceptions? If so, should we put the cleanup/restoring code into a `finally`?

Reply:
I did not check, but that sounds like a good idea. Unfortunately I'm not sure how to do that in C++